app/src/parser_impl.c

```diff
@@ -0,0 +1,873 @@
+{
+    uint8_t binType = 0;
+    uint8_t binLen = 0;
+    CHECK_ERROR(_readUInt8(c, &binType))
+    switch (binType)
+    {
+        case BIN8: {
+            CHECK_ERROR(_readUInt8(c, &binLen))
+            break;
+        }
+        case BIN16:
+        case BIN32: {
+            return parser_msgpack_bin_type_not_supported;
+            break;
+        }
+        default: {
+            return parser_msgpack_bin_type_expected;
+            break;
+        }
+    }
+
+    if(binLen != bufferLen) {
+        return parser_msgpack_bin_unexpected_size;
+    }
+    CHECK_ERROR(_readBytes(c, buff, bufferLen))
+    return parser_ok;
+}
+
+static parser_error_t _readBin(parser_context_t *c, uint8_t *buff, uint16_t *bufferLen, uint16_t bufferMaxSize)
+{
+    uint8_t binType = 0;
+    uint16_t binLen = 0;
+    CHECK_ERROR(_readUInt8(c, &binType))
+    switch (binType)
+    {
+        case BIN8: {
+            uint8_t tmp = 0;
+            CHECK_ERROR(_readUInt8(c, &tmp))
+            binLen = (uint16_t)tmp;
+            break;
+        }
+        case BIN16: {
+            CHECK_ERROR(_readUInt16(c, &binLen))
+            break;
+        }
+        case BIN32: {
+            return parser_msgpack_bin_type_not_supported;
+            break;
+        }
+        default: {
+            return parser_msgpack_bin_type_expected;
+            break;
+        }
+    }
+
+    if(binLen > bufferMaxSize) {
+        return parser_msgpack_bin_unexpected_size;
+    }
+
+    *bufferLen = binLen;
+    CHECK_ERROR(_readBytes(c, buff, *bufferLen))
+    return parser_ok;
+}
+
+parser_error_t _readBool(parser_context_t *c, uint8_t *value)
+{
+    uint8_t tmp = 0;
+    CHECK_ERROR(_readUInt8(c, &tmp))
+    switch (tmp)
+    {
+        case BOOL_TRUE: {
+            *value = 1;
+            break;
+        }
+
+        case BOOL_FALSE: {
+            *value = 0;
+            break;
+        }
+        default: {
+            return parser_msgpack_bool_type_expected;
+            break;
+        }
+    }
+    return parser_ok;
+}
+
+static parser_error_t _readAssetParams(parser_context_t *c, txn_asset_config *asset_config)
+{
+    size_t paramsSize = 0;
+    CHECK_ERROR(_readMapSize(c, &paramsSize))
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_MANAGER))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.manager, sizeof(asset_config->params.manager)))
+    addItem(1);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_RESERVE))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.reserve, sizeof(asset_config->params.reserve)))
+    addItem(2);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_FREEZE))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.freeze, sizeof(asset_config->params.freeze)))
+    addItem(3);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_CLAWBACK))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.clawback, sizeof(asset_config->params.clawback)))
+    addItem(4);
+
+    tx_num_items += 4;
+
+    // These items won't be displayed
```

Comment:
is it ok that any combination of the items below are empty?
what values are we initializing with?
Should the default values be set inside this function too, to ensure consistency?